gstreamer-plugin/git-patch/0001-svthevcenc-Add-svthevc-encoder-element.patch

```diff
@@ -0,0 +1,2059 @@
++  return caps;
++}
++
++static gboolean
++gst_svthevc_enc_sink_query (GstVideoEncoder * enc, GstQuery * query)
++{
++  gboolean res;
++
++  switch (GST_QUERY_TYPE (query)) {
++    case GST_QUERY_ACCEPT_CAPS:{
++      GstCaps *acceptable, *caps;
++
++      acceptable = gst_svthevc_enc_get_supported_input_caps ();
++      gst_query_parse_accept_caps (query, &caps);
++
++      gst_query_set_accept_caps_result (query,
++          gst_caps_is_subset (caps, acceptable));
++      gst_caps_unref (acceptable);
++      res = TRUE;
++    }
++      break;
++    default:
++      res = GST_VIDEO_ENCODER_CLASS (parent_class)->sink_query (enc, query);
++      break;
++  }
++
++  return res;
++}
++
++static GstCaps *
++gst_svthevc_enc_sink_getcaps (GstVideoEncoder * enc, GstCaps * filter)
++{
++  GstCaps *supported_incaps;
++  GstCaps *ret;
++
++  supported_incaps = gst_svthevc_enc_get_supported_input_caps ();
++
++  ret = gst_video_encoder_proxy_getcaps (enc, supported_incaps, filter);
++  if (supported_incaps)
++    gst_caps_unref (supported_incaps);
++  return ret;
++}
++
++static void
++gst_svthevc_enc_class_init (GstSvtHevcEncClass * klass)
++{
++  GObjectClass *gobject_class;
++  GstElementClass *element_class;
++  GstVideoEncoderClass *gstencoder_class;
++
++  gobject_class = G_OBJECT_CLASS (klass);
++  element_class = GST_ELEMENT_CLASS (klass);
++  gstencoder_class = GST_VIDEO_ENCODER_CLASS (klass);
++
++  gobject_class->set_property = gst_svthevc_enc_set_property;
++  gobject_class->get_property = gst_svthevc_enc_get_property;
++  gobject_class->finalize = gst_svthevc_enc_finalize;
++
++  gstencoder_class->set_format = GST_DEBUG_FUNCPTR (gst_svthevc_enc_set_format);
++  gstencoder_class->handle_frame =
++      GST_DEBUG_FUNCPTR (gst_svthevc_enc_handle_frame);
++  gstencoder_class->start = GST_DEBUG_FUNCPTR (gst_svthevc_enc_start);
++  gstencoder_class->stop = GST_DEBUG_FUNCPTR (gst_svthevc_enc_stop);
++  gstencoder_class->flush = GST_DEBUG_FUNCPTR (gst_svthevc_enc_flush);
++  gstencoder_class->finish = GST_DEBUG_FUNCPTR (gst_svthevc_enc_finish);
++  gstencoder_class->getcaps = GST_DEBUG_FUNCPTR (gst_svthevc_enc_sink_getcaps);
++  gstencoder_class->sink_query = GST_DEBUG_FUNCPTR (gst_svthevc_enc_sink_query);
++  gstencoder_class->propose_allocation =
++      GST_DEBUG_FUNCPTR (gst_svthevc_enc_propose_allocation);
++
++  g_object_class_install_property (gobject_class, PROP_INSERT_VUI,
++      g_param_spec_boolean ("insert-vui", "Insert VUI",
++          "Insert VUI NAL in stream",
++          PROP_INSERT_VUI_DEFAULT, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_AUD,
++      g_param_spec_boolean ("aud", "AUD",
++          "Use AU (Access Unit) delimiter", PROP_AUD_DEFAULT,
++          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_HIERARCHICAL_LEVEL,
++      g_param_spec_uint ("hierarchical-level", "Hierarchical Level",
```

Comment:
x264 use this setting, I kept the same in the other implementation.
https://github.com/GStreamer/gst-plugins-ugly/blob/master/ext/x264/gstx264enc.c
 g_param_spec_boolean ("b-pyramid", "B-Pyramid"
what do you think? We use "b-pyramid"?

Reply:
I did the same as the property of the ffmpeg plugin, I think too it would be better to set it equal to gstreamer property name.